utils/determinants_tools.py

```diff
@@ -153,13 +153,33 @@ def csf_to_string(csf, bit_kind_size, size):
             import sys
             sys.exit(1)
 
-    with h5py.File(args.h5path, 'r') as f:
+    open_mode = 'a'
+    with h5py.File(args.h5path, open_mode) as f:
+        alpha_group = 'MultiDet/CI_Alpha'
+        if alpha_group not in f:
+            alpha_group = 'MultiDet/CI_0'
+        beta_group = 'MultiDet/CI_Beta'
+        if beta_group not in f:
+            beta_group = 'MultiDet/CI_1'
 
-        bit_kind_size = f['MultiDet/CI_Alpha'].dtype.itemsize * 8
+        bit_kind_size = f[alpha_group].dtype.itemsize * 8
+        print(f[alpha_group].dtype)
 
         # We sanitize the determinants. Aka we transform any negative integers, into the proper unsigned one.
-        i_alpha= (sanitize(d,bit_kind_size) for d in f['MultiDet/CI_Alpha'])
-        i_beta= (sanitize(d,bit_kind_size) for d in f['MultiDet/CI_Beta'])
+        i_alpha= (sanitize(d,bit_kind_size) for d in f[alpha_group])
+        i_beta= (sanitize(d,bit_kind_size) for d in f[beta_group])
+        if f[alpha_group].dtype == 'int64':
```

Comment:
sorry, in my earlier review I was thinking that this was part of the required QP2->QMCPACK workflow, and that was why I thought we should do the conversion automatically.

I think that if we don't want to handle dets as signed ints in SlaterDetBuilder, then we should provide a tool to do the conversion, but maybe your first way was better (add the `--transform` option to do the transformation, and in SlaterDetBuilder throw an error if the hdf5 uses signed ints (and tell the user how to do the conversion using the python script)).

Reply:
Reverted this. I ended up figuring out how to check the type of a dataset in an hdf5 file.